src/main/java/com/yelp/nrtsearch/server/luceneserver/ShardState.java

```diff
@@ -489,7 +490,7 @@ public synchronized void start() throws Exception {
     }
 
     boolean success = false;
-
+    boolean preloadData = indexState.globalState.configReader.getBoolean(PRELOAD_CONFIG_KEY, true);
```

Comment:
Should we read and store this config parameter in `LuceneServerConfiguration` only? That would make sure all the config options are in one place. We can expose either just preloadData or entire luceneserverconfiguration to ShardState via GlobalState